src/call.cpp

```diff
@@ -2739,6 +2758,19 @@ bool call::process_incoming(char * msg, struct sockaddr_storage *src)
         return false;
     }
 
+    extract_transaction (txn, msg);
+
+    /* If this request came from a different peer than the one who created this
+     * dialog, store off the socket, so that when we send responses on this
+     * transaction we can send them over this socket rather than the default
+     * call_socket.
+     */
+    if (sock && (sock != call_socket) && (txn_sockets[txn] == NULL)) {
```

Comment:
Why does the socket for the current transaction have to be NULL? Is it not possible that we would want to overwrite the socket if we receive a request from a different source?